### PR TITLE
Fixed benchmarks 

### PR DESCRIPTION
**Pull request by zktony**
## Describe your changes
Fixed benchmarks.

### CODE REVIEW

**Comment by zktony** [comment]
Hey @serhii-temchenko, could you please generate weights?

File: pallets/thea-executor/src/mock.rs [hunk 1]
```diff
@@ -140,8 +140,13 @@ parameter_types! {
 
 impl thea::Config for Test {
 	type RuntimeEvent = RuntimeEvent;
+<<<<<<< HEAD
+	type TheaId = thea::ecdsa::AuthorityId;
+	type Signature = thea::ecdsa::AuthoritySignature;
+=======
 	type TheaId = AuthorityId;
 	type Signature = AuthoritySignature;
+>>>>>>> gj/on-chain-validation
```
Comment (serhii-temchenko) [inline pallets/thea-executor/src/mock.rs:149]: Unresolved conflict

File: pallets/ocex/src/lib.rs [hunk 2]
```diff
@@ -93,6 +93,7 @@ mod settlement;
 mod snapshot;
 pub mod storage;
 pub mod validator;
+mod fixtures;
```
Comment (serhii-temchenko) [inline pallets/ocex/src/lib.rs:96]: Should be build only for tests/benchmarks only?

File: pallets/thea/src/benchmarking.rs [hunk 3]
```diff
@@ -1,21 +1,3 @@
...
-// (at your option) any later version.
-//
-// This program is distributed in the hope that it will be useful,
-// but WITHOUT ANY WARRANTY; without even the implied warranty of
-// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
-// GNU General Public License for more details.
-//
-// You should have received a copy of the GNU General Public License
-// along with this program. If not, see <https://www.gnu.org/licenses/>.
-
```
Comment (serhii-temchenko) [inline pallets/thea/src/benchmarking.rs:18]: Please get it back

**Review by serhii-temchenko — CHANGES_REQUESTED** [review]

File: pallets/thea-message-handler/src/lib.rs [hunk 4]
```diff
@@ -52,6 +52,7 @@ pub trait WeightInfo {
 }
 
 pub mod weights;
+mod fixtures;
```
Comment (serhii-temchenko) [inline pallets/thea-message-handler/src/lib.rs:55]: I assume used only in benchmarks, then should build only for benchmark feature.

File: pallets/thea/src/benchmarking.rs [hunk 5]
```diff
@@ -24,38 +6,42 @@ use frame_benchmarking::benchmarks;
 use frame_support::BoundedVec;
 use frame_system::RawOrigin;
 use parity_scale_codec::Decode;
-
+use crate::Pallet as Thea;
```
Comment (serhii-temchenko) [inline pallets/thea/src/benchmarking.rs:9]: Unused?

File: pallets/thea/src/benchmarking.rs [hunk 6]
```diff
@@ -84,10 +70,28 @@ benchmarks! {
...
+		assert!(!active_list.contains(&network));
+	}
 }
 
 #[cfg(test)]
 use frame_benchmarking::impl_benchmark_test_suite;
 
 #[cfg(test)]
-impl_benchmark_test_suite!(Thea, crate::mock::new_test_ext(), crate::mock::Test);
+impl_benchmark_test_suite!(Thea, crate::mock::new_test_ext(), crate::mock::Test);
```
Comment (serhii-temchenko) [inline pallets/thea/src/benchmarking.rs]: Please add empty line at the end of the file.

File: pallets/thea-message-handler/src/fixtures.rs [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+pub(crate) fn produce_authorities<T: Config>(
+) -> BoundedVec<<T as crate::Config>::TheaId, <T as crate::Config>::MaxAuthorities> {
+    let mut authorities: BoundedVec<
+        <T as crate::Config>::TheaId,
+        <T as crate::Config>::MaxAuthorities,
+    > = BoundedVec::with_bounded_capacity(1);
+    let key = <T as crate::Config>::TheaId::decode(&mut PK.as_ref()).unwrap();
+    authorities.try_push(key).unwrap();
+    authorities
+}
```
Comment (serhii-temchenko) [inline pallets/thea-message-handler/src/fixtures.rs]: Please add empty line at the end of the file.

File: pallets/thea-message-handler/src/benchmarking.rs [hunk 8]
```diff
@@ -86,4 +83,4 @@ benchmarks! {
 use frame_benchmarking::impl_benchmark_test_suite;
 
 #[cfg(test)]
-impl_benchmark_test_suite!(TheaMH, crate::mock::new_test_ext(), crate::mock::Test);
+impl_benchmark_test_suite!(TheaMH, crate::mock::new_test_ext(), crate::mock::Test);
```
Comment (serhii-temchenko) [inline pallets/thea-message-handler/src/benchmarking.rs]: Please add empty line at the end of the file.

File: pallets/ocex/src/fixtures.rs [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+    0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0,
+    1, 0, 0, 0, 0, 8, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+    0, 0, 0, 0, 0, 0, 4, 6, 196, 28, 36, 60, 116, 41, 76, 197, 21, 40, 124, 17, 142, 128, 189, 115,
+    168, 219, 199, 151, 158, 208, 8, 177, 131, 105, 116, 42, 17, 129, 26, 0, 0, 0, 0, 0, 0, 0, 0,
+    1, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 184, 239, 65, 246, 134, 193, 91, 100, 60, 198, 195, 116, 249,
+    216, 96, 170, 33, 186, 179, 109, 207, 180, 3, 231, 107, 80, 42, 123, 100, 156, 75, 86, 62, 120,
+    115, 159, 76, 60, 167, 196, 149, 217, 93, 214, 24, 100, 90, 1, 0, 0,
+];
```
Comment (serhii-temchenko) [inline pallets/ocex/src/fixtures.rs]: Please add empty line at the end of the file.

File: pallets/liquidity/src/mock.rs [hunk 10]
```diff
@@ -221,9 +221,9 @@ pub fn new_test_ext() -> sp_io::TestExternalities {
...
 impl<C> frame_system::offchain::SendTransactionTypes<C> for Test
-where
-	RuntimeCall: From<C>,
+	where
+		RuntimeCall: From<C>,
 {
 	type Extrinsic = UncheckedExtrinsic;
 	type OverarchingCall = RuntimeCall;
-}
+}
```
Comment (serhii-temchenko) [inline pallets/liquidity/src/mock.rs]: Please add empty line at the end of the file.

**Review by serhii-temchenko — CHANGES_REQUESTED** [review]

File: pallets/thea-executor/src/benchmarking.rs [hunk 11]
```diff
@@ -18,7 +18,7 @@
 
 #![cfg(feature = "runtime-benchmarks")]
 use super::*;
-
+use crate::{Pallet as TheaExecutor};
```
Comment (serhii-temchenko) [inline pallets/thea-executor/src/benchmarking.rs:21]: Unused?

File: pallets/thea-message-handler/src/benchmarking.rs [hunk 12]
```diff
@@ -19,23 +19,11 @@
...
-	176, 76, 179, 101, 236, 49, 83, 117, 86, 132, 217, 161,
-];
-
-const SIG: [u8; 65] = [
-	246, 101, 246, 156, 149, 156, 74, 60, 188, 84, 236, 77, 232, 165, 102, 241, 137, 124, 100, 143,
-	230, 195, 58, 177, 5, 110, 241, 31, 205, 215, 173, 147, 127, 75, 174, 69, 64, 193, 140, 26, 76,
-	97, 172, 196, 168, 187, 140, 17, 202, 250, 175, 232, 160, 108, 251, 114, 152, 227, 249, 255,
-	186, 113, 211, 53, 0,
-];
+use crate::{Pallet as TheaMH};
```
Comment (serhii-temchenko) [inline pallets/thea-message-handler/src/benchmarking.rs:26]: Unused?

File: pallets/thea/src/benchmarking.rs [hunk 13]
```diff
@@ -24,38 +6,42 @@ use frame_benchmarking::benchmarks;
 use frame_support::BoundedVec;
 use frame_system::RawOrigin;
 use parity_scale_codec::Decode;
-
+use crate::Pallet as Thea;
 // Check if last event generated by pallet is the one we're expecting
 fn assert_last_event<T: Config>(generic_event: <T as Config>::RuntimeEvent) {
 	frame_system::Pallet::<T>::assert_last_event(generic_event.into());
 }
```
Comment (serhii-temchenko) [inline pallets/thea/src/benchmarking.rs:13]: Unused

**Review by serhii-temchenko — CHANGES_REQUESTED** [review]

File: pallets/thea/src/benchmarking.rs (hunk 6 above)
Comment (serhii-temchenko) [reply pallets/thea/src/benchmarking.rs]: Fixed

File: pallets/thea-message-handler/src/fixtures.rs (hunk 7 above)
Comment (serhii-temchenko) [reply pallets/thea-message-handler/src/fixtures.rs]: Fixed

File: pallets/thea-message-handler/src/benchmarking.rs (hunk 8 above)
Comment (serhii-temchenko) [reply pallets/thea-message-handler/src/benchmarking.rs]: Fixed

File: pallets/ocex/src/fixtures.rs (hunk 9 above)
Comment (serhii-temchenko) [reply pallets/ocex/src/fixtures.rs]: Fixed

File: pallets/liquidity/src/mock.rs (hunk 10 above)
Comment (serhii-temchenko) [reply pallets/liquidity/src/mock.rs]: Fixed

**Comment by Gauthamastro** [comment]
Don't merge this, benchmarks is already working in gj/on-chain-validation

**Comment by serhii-temchenko** [comment]
> Don't merge this, benchmarks is already working in gj/on-chain-validation

Got it.

**Comment by serhii-temchenko** [comment]
Closed because of [this](https://github.com/Polkadex-Substrate/Polkadex/pull/836#issuecomment-1643332941) comment.

**Pull request closed without merging**